pkg/network/usm/shared_libraries.go

```diff
@@ -280,40 +277,34 @@ func (w *soWatcher) Start() {
 
 // cleanup removes all registrations
 func (r *soRegistry) cleanup() {
-	r.m.Lock()
-	defer r.m.Unlock()
-
-	for pathID, reg := range r.byID {
-		reg.unregister(pathID)
-	}
+	r.byID.Range(func(key, value any) bool {
+		pathID := key.(pathIdentifier)
+		registry := value.(*soRegistration)
+		registry.unregisterPath(pathID)
+		return true
+	})
 }
 
 // unregister a pid if exists, unregisterCB will be called if his uniqueProcessesCount == 0
 func (r *soRegistry) unregister(pid uint32) {
-	r.m.RLock()
-	_, found := r.byPID[pid]
-	r.m.RUnlock()
+	paths, found := r.byPID.LoadAndDelete(pid)
```

Comment:
CompareAndSwap() would be higher cpu pressure here than the previous read lock implementation